.github/workflows/main.yml

```diff
@@ -2,7 +2,6 @@ name: Build and Test
 
 on:
   push:
-    branches: [master]
```

Comment:
ok, but AFAIR we had the following problem:  when submitting a PR I usually push a tip-<feature>  branch to this repo. I don't want to run CI on both the branch and the PR.  Any other way to "Allow creation of binaries for tags" without removing this ?

Reply:
We can keep both push and pull_request
And maybe add workflow_dispatch to allow manual run

Reply:
problem here is that it's removing the filter "branch == master", so it runs on any branch we push, even those that are pushed just to create a PR

Reply:
> even those that are pushed just to create a PR

As a general development flow, PR's should be done from the developer fork, and not from the main repository, this will turn out in a polluted environment for others developers. If it's a branch that others developers will contribute, makes sense for the CI to run.

Reply:
this also empowers people that are doing their development on their own fork, with that, they are able to have CI running own their custom branches to checkout the binaries and CI. Otherwise, the only way is to creating a PR here.

Reply:
I'm not against running it on branches, only against running it on anything other than the master branch.

well... I don't really care much, so if it's useful for people, ok.